primefaces-selenium/primefaces-selenium-core/src/main/java/org/primefaces/selenium/PrimeSelenium.java

```diff
@@ -152,13 +155,18 @@ public static void goTo(String partialUrl) {
      * @return the URL of the page
      */
     public static String getUrl(AbstractPrimePage page) {
-        DeploymentAdapter deploymentAdapter = ConfigProvider.getInstance().getDeploymentAdapter();
-
         String baseLocation = page.getBaseLocation();
-        if (deploymentAdapter != null) {
+        if (baseLocation == null) {
             baseLocation = getBaseUrl();
         }
-
+        if (baseLocation == null) {
+            String message = "Cannot determine base url. Please either configure " + DEPLOYMENT_BASEURL + " or " +
+                    (ConfigProvider.getInstance().getDeploymentAdapter() != null ?
```

Comment:
introduce ConfigProvider.getInstance().getDeploymentAdapter() as variable, makes it even more readable